web/src/components/FilingsCalendar.test.tsx

```diff
@@ -649,4 +649,58 @@ describe("<FilingsCalendar />", () => {
       expect(screen.queryByText(Config.dashboardDefaults.calendarGridViewButton)).not.toBeInTheDocument();
     });
   });
+
+  describe("calendar list view, view more functionality", () => {
+    function renderCalendarWithEntries(numberOfCalendarEntries: number) {
+      let dueDate: dayjs.Dayjs;
+      let annualReport: TaxFiling;
+      const operateReferences: Record<string, OperateReference> = {};
+      const filings: TaxFiling[] = [];
+
+      for (let i = 0; i < numberOfCalendarEntries; i++) {
+        dueDate = getCurrentDate().add(2, "months").add(i, "day");
+        annualReport = generateTaxFiling({
+          identifier: `annual-report-${i}`,
+          dueDate: dueDate.format(defaultDateFormat),
+        });
+        filings.push(annualReport);
+        operateReferences[`annual-report-${i}`] = generateOperateReference({});
+      }
+
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          operatingPhase: randomElementFromArray(
+            OperatingPhases.filter((obj) => {
+              return obj.displayCalendarType === "FULL" && obj.displayCalendarToggleButton;
+            })
+          ).id,
+        }),
+        taxFilingData: generateTaxFilingData({ filings: filings }),
+        preferences: generatePreferences({ isCalendarFullView: false }),
+      });
+
+      renderFilingsCalendar(operateReferences, userData);
+    }
+
+    it("does not show the view more button if 5 or less events are in the calendar", () => {
+      renderCalendarWithEntries(3);
+      expect(screen.getAllByTestId("calendar-list-entry")).toHaveLength(3);
+      expect(screen.queryByText("View More")).not.toBeInTheDocument();
+    });
+
+    it("shows the view more button if more than 5 events are in the calendar", () => {
+      renderCalendarWithEntries(7);
+      expect(screen.getAllByTestId("calendar-list-entry")).toHaveLength(5);
+      expect(screen.getByText("View More")).toBeInTheDocument();
+    });
+
+    it("shows 5 more events when the view more button is clicked", () => {
+      renderCalendarWithEntries(12);
+      expect(screen.getAllByTestId("calendar-list-entry")).toHaveLength(5);
+      fireEvent.click(screen.getByText("View More"));
+      expect(screen.getAllByTestId("calendar-list-entry")).toHaveLength(10);
+      fireEvent.click(screen.getByText("View More"));
+      expect(screen.getAllByTestId("calendar-list-entry")).toHaveLength(12);
```

Comment:
I think there's a missing final expectation where the View More button goes away when there are no more events to show

Reply:
Yeah good catch